LFPy/eegmegcalc.py

```diff
@@ -0,0 +1,765 @@
+        pot_sum = np.sum([c*lpmv(1, i, np.cos(theta)) for c,i in zip(consts,n)])
+        return pot_sum
+
+    def _calc_vn(self, n):
+        r_const = (self.r34 ** n - self.r43 ** (n + 1)) / ((n + 1) / n * self.r34 ** n + self.r43 ** (n + 1))
+        v = (n / (n + 1) * self.sigma34 - r_const) / (self.sigma34 + r_const)
+        return v
+
+    def _calc_yn(self, n):
+        vn = self._calc_vn(n)
+        r_const = (n / (n + 1) * self.r23 ** n - vn * self.r32 ** (n + 1)) / (self.r23 ** n + vn * self.r32 ** (n + 1))
+        y = (n / (n + 1) * self.sigma23 - r_const) / (self.sigma23 + r_const)
+        return y
+
+    def _calc_zn(self, n):
+        yn = self._calc_yn(n)
+        z = (self.r12 ** n - (n + 1) / n * yn * self.r21 ** (n + 1)) / (self.r12 ** n + yn * self.r21 ** (n + 1))
+        return z
+
+    def _calc_c1n(self, n):
+        zn = self._calc_zn(n)
+        c = ((n + 1) / n * self.sigma12 + zn) / (self.sigma12 - zn) * self.rz1**(n+1)
+        return c
+
+    def _calc_c2n(self, n):
+        yn = self._calc_yn(n)
+        c1 = self._calc_c1n(n)
+        c2 = (c1 + self.rz1**(n+1)) / (self.r12 ** n + yn * self.r21 ** (n + 1))
+        return c2
+
+    def _calc_d2n(self, n, c2):
+        yn = self._calc_yn(n)
+        d2 = yn * c2
+        return d2
+
+    def _calc_c3n(self, n):
+        vn = self._calc_vn(n)
+        c2 = self._calc_c2n(n)
+        d2 = self._calc_d2n(n, c2)
+        c3 = (c2 + d2) / (self.r23 ** n + vn * self.r32 ** (n + 1))
+        return c3
+
+    def _calc_d3n(self, n, c3):
+        vn = self._calc_vn(n)
+        d3 = vn * c3
+        return d3
+
+    def _calc_c4n(self, n):
+        c3 = self._calc_c3n(n)
+        d3 = self._calc_d3n(n, c3)
+        c4 = (n + 1) / n * (c3 + d3) / ((n + 1) / n * self.r34 ** n + self.r43 ** (n + 1))
+        return c4
+
+    def _calc_d4n(self, n, c4):
+        d4 = n / (n + 1) * c4
+        return d4
+
+
+class InfiniteVolumeConductor(object):
+    """
+    Main class for computing potentials with current dipole approximation.
+
+    Parameters
+    __________
+    sigma : float [S/cm]
+            Electrical conductivity in extracellular space.
+    Examples
+    ________
+    Computing the potential from dipole moment valid in the far field limit.
+    Theta correspond to the dipole alignment angle from the vertical z-axis.
+    >>> import LFPy
+    >>> import numpy as np
+    >>> inf_model = LFPy.InfiniteVolumeConductor(sigma=0.3)
+    >>> P = np.array([[10., 10., 10.]])
+    >>> Phi_P, theta = inf_model.get_dipole_potential(P,
+    >>>                                               x=np.array([1000]),
+    >>>                                               y=np.array([0]),
+    >>>                                               z=np.array([5000]))
+
+    """
+
+    def __init__(self, sigma=0.3):
+        "Initialize class InfiniteVolumeConductor"
+        self.sigma = sigma
+
+    def get_dipole_potential(self, P, x, y, z):
```

Comment:
I think we should use a simpler implementation and definition of the current-dipole moment potential here, also less ambiguous and likely faster to compute with `np.dot()`:

Electric potential of current-dipole moment in infinite, homogeneous and linear extracellular medium:
	\begin{equation}
	\phi{(\mathbf{R})} = \frac{\mathbf{p} \cdot \mathbf{R}}{4 \pi \sigma R^3}~,
	\end{equation}
	where $\mathbf{R}=\mathbf{r}-\mathbf{r}'$ is the vector between dipole location $\mathbf{r}'$ and measurement location $\mathbf{r}$, and $R=|\mathbf{R}|$.